src/components/Map/Layers/TileLayer.jsx

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from 'react';
+import OLTileLayer from 'ol/layer/Tile';
+
+function TileLayer({
+  map, source, zIndex = 0, opacity = 1,
+}) {
+  const [tileLayer, setTileLayer] = useState(false);
+
+  useEffect(() => {
+    if (!map) return undefined;
+
+    const tileRasterLayer = new OLTileLayer({
+      source,
+      zIndex,
+    });
+    map.addLayer(tileRasterLayer);
+    tileRasterLayer.setZIndex(zIndex);
+    tileRasterLayer.setOpacity(opacity);
+
+    setTileLayer(tileRasterLayer);
+
+    return () => {
+      if (map) {
+        map.removeLayer(tileRasterLayer);
+      }
+    };
+  }, [map, JSON.stringify(source.urls)]);
```

Comment:
If we make source immutable, we do not need to use `JSON.stringify(source.urls)`
We can just use `source.urls`

Reply:
Also, why do we not have the `source` object a dependency?

Reply:
this is working ok using the approach above - if i remove JSON.stringify or when using just 'source' as the dependency, the layer redraws when dragging opacity slider.